website/project/model.py

```diff
@@ -2749,6 +2749,8 @@ def embargo_registration(self, user, end_date, for_existing_registration=False):
         )
         # Embargo record needs to be saved to ensure the forward reference Node->Embargo
         self.embargo = embargo
+        if self.is_public:
+            self.set_privacy('private', Auth(user))
```

Comment:
Take a careful look at `set_privacy` to make sure. I believe there are some addon hooks that are called there.